features/remotepollerng/daemon/src/main/java/org/opennms/netmgt/remotepollerng/RemotePollerd.java

```diff
@@ -321,25 +327,52 @@ public void saveResponseTimeData(final String locationName, final OnmsMonitoredS
-        latencyResource.addAttribute(new LatencyCollectionAttribute(latencyResource,
-                latencyType, dsName, responseTime));
+        // Prefer ds-name over "response-time" for primary response-time value
+        final Map<String, Number> properties = Maps.newHashMap(pollStatus.getProperties());
+        if (!properties.containsKey(dsName) && properties.containsKey(PollStatus.PROPERTY_RESPONSE_TIME)) {
+            properties.put(dsName, properties.get(PollStatus.PROPERTY_RESPONSE_TIME));
+            properties.remove(PollStatus.PROPERTY_RESPONSE_TIME);
+        }
 
-        ServiceParameters params = new ServiceParameters(Collections.emptyMap());
-        CollectionSetVisitor persister = persisterFactory.createPersister(params, repository, false, true, true);
+        // Build collection agent
+        final CollectionAgentDTO agent = new CollectionAgentDTO();
+        agent.setAddress(monSvc.getIpAddress());
+        agent.setForeignId(monSvc.getForeignId());
+        agent.setForeignSource(monSvc.getForeignSource());
+        agent.setNodeId(monSvc.getNodeId());
+        agent.setNodeLabel(monSvc.getIpInterface().getNode().getLabel());
+        agent.setLocationName(locationName);
+        agent.setStorageResourcePath(ResourcePath.get(LocationUtils.isDefaultLocationName(residentLocationName)
+                                                      ? ResourcePath.get()
+                                                      : ResourcePath.get(ResourcePath.sanitize(residentLocationName)),
+                                                      InetAddressUtils.str(monSvc.getIpAddress())));
+        agent.setStoreByForeignSource(false);
+
+        // Create collection set from response times as gauges and persist
+        final CollectionSetBuilder collectionSetBuilder = new CollectionSetBuilder(agent);
+        final RemoteLatencyResource resource = new RemoteLatencyResource(locationName, InetAddressUtils.str(monSvc.getIpAddress()), svcName);
+        for (final Map.Entry<String, Number> e: properties.entrySet()) {
+            final String key = PollStatus.PROPERTY_RESPONSE_TIME.equals(e.getKey())
+                               ? dsName
+                               : e.getKey();
+
+            collectionSetBuilder.withGauge(resource, rrdBaseName, key, e.getValue());
+        }
 
-        SingleResourceCollectionSet collectionSet = new SingleResourceCollectionSet(latencyResource, new Date());
-        collectionSet.setStatus(CollectionStatus.SUCCEEDED);
-        collectionSet.visit(persister);
+        collectionSetBuilder.build()
+                            .visit(this.persisterFactory.createPersister(new ServiceParameters(Collections.emptyMap()),
+                                                                         repository,
+                                                                         false,
+                                                                         true,
```

Comment:
Do we need to enforce storeByGroup here?

Reply:
Yes. Response Times are always stored by group. Allowing to configure this would require to add another level in folder hierarchy, because we have to store multiple "response" tracks.

Reply:
Yes. Response Times are always stored by group. Allowing to configure this would require to add another level in folder hierarchy, because we have to store multiple "response" tracks.